### PR TITLE
Add equal & isString

### PR DESCRIPTION
**Pull request by cagataycali**
Equal & isString method added

### CODE REVIEW

**Comment by Acanguven** [comment]
# [Codecov](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=h1) Report
> Merging [#15](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=desc) into [master](https://codecov.io/gh/Acanguven/rapidash/commit/02b0defe56d16c04e774bd9b9d8cd3ffb228a585?src=pr&el=desc) will **not change** coverage.
> The diff coverage is `100%`.

[Impacted file tree graph](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=tree)

```diff
@@          Coverage Diff          @@
##           master    #15   +/-   ##
=====================================
  Coverage     100%   100%           
=====================================
  Files           7      8    +1     
  Lines          68    105   +37     
  Branches        8     25   +17     
=====================================
+ Hits           68    105   +37
```


| [Impacted Files](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/functions/equal.ts](https://codecov.io/gh/Acanguven/rapidash/pull/15/diff?src=pr&el=tree#diff-c3JjL2Z1bmN0aW9ucy9lcXVhbC50cw==) | `100% <100%> (ø)` | |
| [src/functions/index.ts](https://codecov.io/gh/Acanguven/rapidash/pull/15/diff?src=pr&el=tree#diff-c3JjL2Z1bmN0aW9ucy9pbmRleC50cw==) | `100% <100%> (ø)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=footer). Last update [02b0def...e5bf979](https://codecov.io/gh/Acanguven/rapidash/pull/15?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/functions/equal.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
+import { SolutionBuilder } from '../solution';
+
+export type definition = (a: any, b: any) => boolean;
+
+export const solution = new SolutionBuilder<definition>('equal');
+
+// Original repository: https://github.com/epoberezkin/fast-deep-equal
+// Modified & bugs fixed by @cagataycali
+function equal(a: any, b: any): boolean {
```
Comment (Acanguven) [inline src/functions/equal.ts]: Moving this into fn() would be better for convention.

File: src/functions/equal.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  ]);
+
+/**
+ * Solutions
+ * Provide your solutions below
+ */
+
+solution
+  .owner('cagataycali')
+  .method('Like a boss.')
```
Comment (Acanguven) [inline src/functions/equal.ts]: At least some technical terms :)

**Comment by yavuzkoca** [comment]
IMO you should separate the solutions into different pull requests

**Comment by cagataycali** [comment]
You’re right, but we talked with author and this pr happened

**Pull request merged**